frontend/src/components/TraineesAutocomplete/TraineesAutocomplete.tsx

```diff
@@ -0,0 +1,58 @@
+import useSWR from 'swr';
+import { TraineeFindAllResponseDto } from '@osk/shared';
+import { useState } from 'react';
+import { addQueryParams } from '../../utils/addQueryParams';
+import { GeneralAPIError } from '../GeneralAPIError/GeneralAPIError';
+import { FAutocomplete } from '../FAutocomplete/FAutocomplete';
+import { useDebounce } from '../../hooks/useDebounce/useDebounce';
+
+interface TraineesAutocompleteProps {
+  required: boolean;
+}
+
+export const TraineesAutocomplete = ({
+  required,
+}: TraineesAutocompleteProps) => {
+  const [inputValue, setInputValue] = useState<string>('');
+  const [savedInputValue, setSavedInputValue] = useState<string | null>(null);
+  const debouncedValue = useDebounce(inputValue);
+  const { data, error, isValidating } = useSWR<TraineeFindAllResponseDto>(
+    debouncedValue
+      ? addQueryParams('/api/trainees', {
+          filters: { searchedPhrase: savedInputValue ?? debouncedValue },
+        })
+      : null,
+  );
+
+  const traineesOptions =
+    data?.trainees?.map((option) => {
+      return {
+        label: `${option.user.firstName} ${option.user.lastName} (tel: ${option.user.phoneNumber})`,
+        id: option.id,
+      };
+    }) ?? [];
+
+  if (error) {
+    return <GeneralAPIError />;
+  }
+
+  return (
+    <FAutocomplete
+      onInputChange={(newValue, reason) => {
+        if (reason === 'reset') {
+          setSavedInputValue(inputValue);
+        } else {
+          setSavedInputValue(null);
+        }
+        setInputValue(newValue ?? '');
+      }}
+      inputValue={inputValue}
+      options={traineesOptions}
+      loading={isValidating}
+      label="Kursant"
+      name="traineeId"
+      id="traineeId"
```

Comment:
I'd personally add those as a prop to this comonent, so that potentially those could be parametrized from the outside